includes/Checker/Checks/Enqueued_Styles_Scope_Check.php

```diff
@@ -0,0 +1,247 @@
+
+		return function() use ( $orig_scripts ) {
+			if ( is_null( $orig_scripts ) ) {
+				unset( $GLOBALS['wp_styles'] );
+			} else {
+				$GLOBALS['wp_styles'] = $orig_scripts;
+			}
+
+			$this->restore_globals();
+		};
+	}
+
+	/**
+	 * Returns an array of shared preparations for the check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Returns a map of $class_name => $constructor_args pairs. If the class does not
+	 *               need any constructor arguments, it would just be an empty array.
+	 */
+	public function get_shared_preparations() {
+		$demo_posts = array_map(
+			static function( $post_type ) {
+				return array(
+					'post_title'   => "Demo {$post_type} post",
+					'post_content' => 'Test content',
+					'post_type'    => $post_type,
+					'post_status'  => 'publish',
+				);
+			},
+			$this->get_viewable_post_types()
+		);
+
+		return array(
+			Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+		);
+	}
+
+	/**
+	 * Runs the check on the plugin and amends results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results to amend and the plugin context.
+	 */
+	public function run( Check_Result $result ) {
+		$this->run_for_urls(
+			$this->get_urls(),
+			function () use ( $result ) {
+				$this->check_url( $result );
+			}
+		);
+
+		if ( ! empty( $this->plugin_styles ) ) {
+			$url_count = count( $this->get_urls() );
+			foreach ( $this->plugin_styles as $plugin_style ) {
+				if ( isset( $plugin_style['count'] ) && ( $url_count === $plugin_style['count'] ) ) {
+					$result->add_message(
+						false,
+						__( 'This style is being loaded in all contexts.', 'plugin-check' ),
+						array(
+							'code' => 'EnqueuedStylesScope.StyleLoadedInAllContext',
+							'file' => $plugin_style['path'],
+						)
+					);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Gets the list of URLs to run this check for.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of URL strings (either full URLs or paths).
+	 *
+	 * @throws Exception Thrown when a post type URL cannot be retrieved.
+	 */
+	protected function get_urls() {
+		$urls = array( home_url( '/' ) );
+
+		foreach ( $this->get_viewable_post_types() as $post_type ) {
+			$args = array(
+				'posts_per_page'         => 1,
+				'post_type'              => $post_type,
+				'post_status'            => array( 'publish', 'inherit' ),
+				'ignore_sticky_posts'    => true,
+				'no_found_rows'          => true,
+				'lazy_load_term_meta'    => false,
+				'update_post_meta_cache' => false,
+				'update_post_term_cache' => false,
+			);
+
+			$the_query = new \WP_Query( $args );
+
+			if ( $the_query->have_posts() ) {
+				while ( $the_query->have_posts() ) {
+					$the_query->the_post();
+
+					$urls[] = get_permalink();
+
+					// Add post category link.
+					$id = get_the_ID();
+					if ( 'post' === get_post_type( $id ) ) {
+						$category = get_the_category( $id );
+						$urls[]   = get_category_link( $category[0]->term_id );
+					}
```

Comment:
I think we should test extra taxonomy type as well here. Just testing categories seems a little random.